packages/anoncreds/src/updates/0.3.1-0.4/__tests__/proofExchangeRecord.test.ts

```diff
@@ -0,0 +1,81 @@
+import { ProofExchangeRecord, ProofRepository, JsonTransformer, ProofState, ProofRole } from '../../../../../core/src'
+import { Agent } from '../../../../../core/src/agent/Agent'
+import { getAgentConfig, getAgentContext, mockFunction } from '../../../../../core/tests'
+import * as testModule from '../proofExchangeRecord'
+
+const agentConfig = getAgentConfig('AnonCreds Migration - Proof Exchange Record - 0.3.1-0.4.0')
+const agentContext = getAgentContext()
+
+jest.mock('../../../../../core/src/modules/proofs/repository/ProofRepository')
+const ProofRepositoryMock = ProofRepository as jest.Mock<ProofRepository>
+const proofRepository = new ProofRepositoryMock()
+
+jest.mock('../../../../../core/src/agent/Agent', () => {
+  return {
+    Agent: jest.fn(() => ({
+      config: agentConfig,
+      context: agentContext,
+      dependencyManager: {
+        resolve: jest.fn(() => proofRepository),
+      },
+    })),
+  }
+})
+
+// Mock typed object
+const AgentMock = Agent as jest.Mock<Agent>
+
+describe('0.3.1-0.4.0 | AnonCreds Migration | Proof Exchange Record', () => {
+  let agent: Agent
+
+  beforeEach(() => {
+    agent = new AgentMock()
+  })
+
+  describe('migrateProofExchangeRecordToV0_4()', () => {
+    it('should fetch all records and apply the needed updates ', async () => {
+      const records: ProofExchangeRecord[] = [getProofRecord({})]
+
+      mockFunction(proofRepository.getAll).mockResolvedValue(records)
+
+      await testModule.migrateProofExchangeRecordToV0_4(agent)
+
+      expect(proofRepository.getAll).toHaveBeenCalledTimes(1)
+      expect(proofRepository.update).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  describe('migrateRole()', () => {
+    // according to: https://github.com/hyperledger/aries-rfcs/blob/main/features/0037-present-proof/README.md#states
+    genMigrateRoleTests(agent, ProofState.RequestSent, ProofRole.Verifier)
+    genMigrateRoleTests(agent, ProofState.ProposalReceived, ProofRole.Verifier)
+    genMigrateRoleTests(agent, ProofState.RequestReceived, ProofRole.Verifier)
+    genMigrateRoleTests(agent, ProofState.PresentationReceived, ProofRole.Verifier)
+
+    genMigrateRoleTests(agent, ProofState.RequestReceived, ProofRole.Prover)
+    genMigrateRoleTests(agent, ProofState.ProposalSent, ProofRole.Prover)
+    genMigrateRoleTests(agent, ProofState.PresentationSent, ProofRole.Prover)
+  })
```

Comment:
This is not testing state done, which was still a fixme in the previous PR. I think same for credential